src/App.tsx

```diff
@@ -8,7 +8,65 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
+interface TodosFilterOptions {
+  todos: Todo[];
+  query: string;
+  filterBy: FilterType;
+}
+
+const getVisibleTodos = (options: TodosFilterOptions): Todo[] => {
+  const { todos, query, filterBy } = options;
+
+  let visibleTodos = [...todos];
+
+  if (query) {
+    const normalizedQuery = query.trim().toLowerCase();
+
+    visibleTodos = visibleTodos
+      .filter((todo) => todo.title
+        .toLowerCase()
+        .includes(normalizedQuery));
+  }
+
+  if (filterBy) {
+    visibleTodos = visibleTodos.filter((todo) => {
+      switch (filterBy) {
+        case FilterType.Active:
+          return !todo.completed;
+        case FilterType.Completed:
+          return todo.completed;
+        default:
+          return FilterType.All;
+      }
+    });
+  }
+
+  return visibleTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.All);
+
+  const visibleTodos = getVisibleTodos({ todos, query, filterBy });
+
+  const shouldDisplayLoader = isLoading;
+  const shouldDisplayTodoList = !isLoading && todos.length > 0;
```

Comment:
```suggestion
  const shouldDisplayTodoList = !isLoading && todos.length > 0;
```
There is no need to create a variable for a single condition. It makes sense with two and more conditions